CMakeLists.txt

```
@@ -1143,6 +1143,19 @@ if(HPX_WITH_NETWORKING)
     ADVANCED
   )
 
+  hpx_option(
+    HPX_WITH_PARCELPORT_GASNET BOOL "Enable the GASNET based parcelport." OFF
+    CATEGORY "Parcelport"
+  )
+  if(HPX_WITH_PARCELPORT_GASNET)
+    hpx_add_config_define(HPX_HAVE_PARCELPORT_GASNET)
+    hpx_option(
+      HPX_WITH_PARCELPORT_GASNET_CONDUIT STRING
+      "Define which conduit to use for the gasnet parcelport" "udp"
+      STRINGS "smp;udp;mpi;ofi;ucx;ibv;aries"
+    )
+  endif()
```

Should we enable aries only on cray platforms? Not sure if there is a way to know...
```suggestion
  if(HPX_WITH_PARCELPORT_GASNET)
    if(...is cray aries available)
      set(additional_conduits ";aries"
    endif()
    hpx_add_config_define(HPX_HAVE_PARCELPORT_GASNET)
    hpx_option(
      HPX_WITH_PARCELPORT_GASNET_CONDUIT STRING
      "Define which conduit to use for the gasnet parcelport" "udp"
      STRINGS "smp;udp;mpi;ofi;ucx;ibv" ${additional_conduits}
    )
  endif()
```

---

@hkaiser point taken! The GASnet team has a fair amount of [autconf scripting](https://bitbucket.org/berkeleylab/gasnet/src/stable/configure.in) implemented to detect Cray platforms and interconnects. Probably best to leave that for a later date.

---

Sure, no problem. Will there be at least some error message if somebody selects a conduit that is not available?